src/Orchard.Web/Modules/Orchard.OpenId/Routes/OpenId.cs

```diff
@@ -44,6 +44,57 @@ public class OpenIdRoutes : IRouteProvider {
                 new RouteDescriptor {
                     Priority = 10,
                     Route = new Route(
+                        "Users/Account/LogOn",
```

Comment:
No need to replace the routes, once the user sets the settings correctly of one of the enabled OpenId features the app will register the routes automatically after the first restart.
This was intended, no need to override the routes always, only override when the replacement is valid.
I think a more proper solution will be notifying the user that an app restart is required as @sebastienros earlier suggested.